README
Webacula - web interface of a Bareos/Bacula backup system

Version: 5.5.2.x

http://webacula.sourceforge.net/
http://github.com/tim4dev/webacula

https://github.com/bareos/contrib-webacula/ is a fork of the Webacula project,
to adapt this software to be usable with Bareos (a Bacula fork).
It is based on Webacula version 5.5.2. The version for the Bareos are appended to this version (5.5.2.x).
The default branch inside this fork is 
  * bareos-master
Other "bareos-*" branches may follow.

Webacula basic features:
  * Bareos/Bacula and Webacula ACLs (Access Control Lists) implemented. ACLs stored in Bareos/Bacula database in Webacula tables
  * Full supported MySQL, PostgreSQL and Sqlite databases
  * Run Job
  * Restore all files or selected files from Job by JobId
  * Restore the most recent backup for a client
  * Restore backup for a client before a specified time
  * Show Timeline for Jobs
  * Mount, umount Storages
  * Show Jobs which executed with errors
  * Show a condition of your Volumes
  * Show scheduled Jobs (at 24 hours forward)
  * Show all running Jobs
  * Show terminated Jobs (executed in last 24 hours)
  * Search Jobs on several conditions
  * Web (ajax) terminal to Bareos/Bacula Console
  * The detailed information on Pools, Volumes, Storages and Clients
  * Logbook. Logbook - is simple electronic journal of backups. 
    Records in logbook are insert, modify, delete manually by the operator. 
    Records can contain links (usual web-links on which it is possible to click) to Bacula Jobs or 
    links to other records of this logbook. 
    In logbook insert records, for example, about Jobs, failures of the equipment, supernumerary situations, etc. 
    Logbook stored in a separate database "webacula", now it is supported MySQL, PostgreSQL and Sqlite.
  * RSS feed
  * Multilingual support - standard gettext (*.mo) files. Currently supported languages: English, Czech, French, German, Italian, Portuguese Brazil, Russian, Spanish.
  * For web designers : now very easy to create new skins



-------

Copyright 2007, 2008, 2009, 2010, 2011, 2012 Yuri Timofeev <tim4dev@gmail.com>

This program is free software: you can redistribute it and/or modify it under the terms of
the GNU General Public License as published by the Free Software Foundation, either version 3 of the License,
or (at your option) any later version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
See the GNU General Public License for more details.

You should have received a copy of the GNU General Public License along with this program.
If not, see http://www.gnu.org/licenses/


ChangeLog
Webacula ChangeLog
------------------

=== 5.5.2.5 (We Nov 14 2013)
    added webacula-config to automatically configure webacula and the database
        add a own database user "webacula" for accessing the bacula/bareos database
        removed db.conf, because data is taken from config.ini
        no default DB
    don't preset the timezone. Take system default.
    added sudo configuration
    prepare handling of databases on differnet hosts
    set default http configuration to APPLICATION_ENV development. 
      Otherwise configuration errors are to hard to find.
    Allow access from everywhere, not only localhost.
    Add SPEC file for obs (openSUSE, SLES, RHEL, CentOS)

=== 5.5.2.4 (We Oct 16 2013)
    Changed Default DB-Params to PGSQL
    Changed SQLite settings to bareos
    Added the new bconsole commands to the web bconsole


=== 5.5.2.3 (Tue Jun 11 2013)
    Changed Parameters to work with bareos
    Fixed catalog version problem
    Fixed depricated values in table webacula_php_session for id (increase to 128)

=== 5.5.2 (Sun Oct 30 2011)
    Show human readable short Job description instead (or both) of Bacula Job names (see config.ini for details).
    Show Bacula Job resource record (bconsole> show job="xxx").
    Small bugfixes.

=== 5.5.1 (Sat Sep 10 2011)
    Small bugfixes.

=== 5.5.0
    Implemented Bacula ACLs : JobACL, ClientACL, StorageACL, PoolACL, FileSetACL, WhereACL.
    Implemented Webacula ACLs - access to main menu.
    Move all Webacula tables to Bacula database. Now a separate database is not needed.
    Greatly increased performance when restoring the file list : implemented receive a list of files to restore from temporary table (trac ticket #40).
    Remove 'tmpdir' from config.ini. Now 'tmpdir' is not needed.
    Fix bug #3151906 -- "Single" files missing from "Select Files for Restore".
    Documentation to install Webacula on Ubuntu/Debian based.
    PHP session is stored in the table.

=== 5.0.3 (Mon Jan 24 2011)
    Fix bug #3151906 -- "Single" files missing from "Select Files for Restore" 

=== 5.0.2 (Wed Aug 11 2010)
    Fresh Portuguese Brazil translation by Samuel Rios Carvalho

=== 5.0.1 (Thu May 12 2010)
    fix bug #3000461 Restore selected files

=== 5.0 (Thu Feb 20 2010)
    Important : required Bacula 5.0 or later.
    Important : now database 'webacula' NOT required for Restore Jobs. Now create a tree of files to Job Restore is faster at 30%--50%
    Add few options into restore pages.
    Add Spanish translation by Efren Bravo <efrenba@yahoo.es>
    Implements "Reviewed" feature: if the Job completed with an error, then you can click on the link "Written in Logbook" and "Reviewed" this Job. Then don't show this Job in list "Jobs with errors" in future. See also trac ticket #43.

=== 3.5 (Tue Feb 16 2010)
   Job->Search->File name (LIKE, REGEX) and restore single file.
   Support mount/unmount Autochanger.
   Display Autochanger content.
   Extend the "Run Job" page by options to modify a Job before running : Pool, Client, FileSet, Storage, Level, Start Time.
   Add links to the Jobs displayed in a Timeline graphic (use imagemap).
   Support multiple Restore Job resources.
   Take into consideration new Job fields : PriorJobId (link to prior Job Record when migrated),
PurgedFiles (set when all File records purged). See 'developers/Database_Tables.html' for details.
   Improved User Interface.
   Fix bugs.

=== 3.4.1
   Fix bug #545471 in bugzilla RedHat.

=== 3.4  (Wed Dec  2  2009)
   Add "Select full restore to a specified JobId".
   Add web/ajax terminal to Bacula Console.
   Add on "List of Running Jobs" the view of "joblog", so is possible to check the running session.
   Add "List Volumes likely to need replacement from age or errors", like bconsole -> query -> 16
   Fix bug #2891270 : ereg* deprecated in php >= 5.3
   Add 'Job description' feature to Logbook.
   Add 'Cancel JobId'.
   Improved User Interface.

=== 3.3
   Now main page (index, "Desktop" menu) does not show a message like "Not found xyz".
   Pools, Volumes operations.
   Now the Zend Framework need to be downloaded separately, from http://framework.zend.com/
   New ini-parameter "days_to_show_jobs_with_errors". As many days to show those Jobs that are completed with errors (default = 7).

=== 3.2
   Improved design.
   Fixed few bugs.
   Restructuring of source code : Move all SQL code from controllers to models.
   Restructuring of source code : use Zend_Layout. Now very easy to create new skins. 
Important information for web designers described in file 4CONTRIBUTORS.

=== 3.1 release (2009-09-16)
   Add full PostgreSQL support (restore files, logbook, etc).
   Add full Sqlite support (restore files, logbook, etc).
   Rewrite drawFileTreeAction().
   Add regress testing procedures.
   More bugfixes.

=== 3.0
   Fixed several unexpected bugs.
   Remove prototype.js. Use json2.js and jquery.js instead.
   Replaced by Calendar to jQuery Datepicker widget.
   Conversion to a standard type of language files.
   Add parameter 'mainmenu_floating' in config.ini. A floating menu stays visible even if you scroll down a web page.
   Add parameter 'head_title' in config.ini
   Add RSS feed
   Add "Restore the most recent backup for a client"
   Add "Restore backup for a client before a specified time"
   New parameter in config.ini : "ttl_restore_session"
   Add exceptions handler
   Improved design
   Add docs/check_system_requirements.php
   Renumbering versions. Now major version number of WeBacula will match the version number of Bacula.
   Create branch-1.38-2.x for the older versions webacula (they are working with bacula 1.38...2.x)
   Up to Zend Framework 1.8.3

=== 1.5
   Add Restore the most recent backup for a client
   Add Restore backup for a client before a specified time 
   Add full text search to LogBook
   Fix bug (Zend/Paginator/Adapter/DbSelect.php) http://framework.zend.com/issues/browse/ZF-5620
   Fix Logbook find by Id record

install/PostgreSql/20_acl_make_tables.sh
#!/bin/bash
#
# Script to create webacula tables
#

webacula_root_pwd=${webacula_root_pwd:-"CHANGEME"}
CMD=${CMD:-`/usr/sbin/webacula-config get_sql_cmd`}

printf "CMD: $CMD\n"
eval $CMD <<END-OF-DATA

SET client_min_messages=WARNING;

CREATE TABLE webacula_users (
    id       SERIAL NOT NULL,
    login    varchar(50) UNIQUE NOT NULL,
    pwd      varchar(50) NOT NULL,
    name     varchar(150),
    email    varchar(50),
    create_login timestamp without time zone,
    last_login timestamp without time zone,
    last_ip  varchar(40),
    active   integer,
    role_id  integer NOT NULL,
    PRIMARY KEY (id)
);
CREATE INDEX webacula_users_idx1 ON webacula_users (login);


CREATE TABLE webacula_roles (
    id      SERIAL NOT NULL,
    order_role  integer not null DEFAULT 1,
    name    varchar(50) UNIQUE not null,
    description TEXT,
    inherit_id  integer,
    primary key (id)
);


CREATE TABLE webacula_resources (
    id       SERIAL NOT NULL,
    dt_id    integer,
    role_id  integer,
    primary key (id)
);


CREATE TABLE webacula_dt_resources (
    id      SERIAL NOT NULL,
    name    varchar(50) UNIQUE not null,
    description TEXT NOT NULL,
    primary key (id)
);



INSERT INTO webacula_roles (id, name, description) VALUES (1, 'root_role', 'Default built-in superuser role');
INSERT INTO webacula_users (id, login, pwd, name, active, create_login, role_id)
    VALUES (1000, 'root', MD5('$webacula_root_pwd'), 'root', 1, NOW(), 1);

INSERT INTO webacula_roles (id, name, description) VALUES (2, 'operator_role', 'Typical built-in role for backup operator');

INSERT INTO webacula_resources (dt_id, role_id) VALUES
    (10,2),
    (20,2),
    (30,2),
    (40,2),
    (50,2),
    (60,2),
    (70,2),
    (80,2),
    (90,2),
    (100,2),
    (110,2),
    (120,2),
    (130,2),
    (140,2),
    (150,2),
    (160,2);

-- Controller names only
INSERT INTO webacula_dt_resources (id, name, description) VALUES
    (10, 'index',     'Home page'),
    (20, 'bconsole',  'Webacula bconsole'),
    (30, 'client',    'Menu Client'),
    (40, 'director',  'Menu Director'),
    (50, 'file',      'List Files for JobId'),
    (60, 'job',       'Menu Job'),
    (70, 'log',       'View console log for Job'),
    (80, 'pool',      'Menu Pool'),
    (90, 'restorejob','Menu Restore Job'),
    (100,'storage',   'Menu Storage'),
    (110,'volume',    'Menu Volume'),
    (120,'wbjobdesc', 'Menu Job Descriptions'),
    (130,'wblogbook', 'Menu Logbook'),
    (140,'help',      'Menu Help'),
    (150,'feed',      'RSS feed'),
    (160,'chart',     'Timeline chart'),
    (500,'admin',     'Menu Administrator');



-- Bacula ACLs


CREATE TABLE webacula_client_acl (
    id        SERIAL NOT NULL,
    name      varchar(127),
    order_acl integer,
    role_id   integer,
    PRIMARY KEY (id)
);
CREATE UNIQUE INDEX webacula_client_acl_idx1 ON webacula_client_acl (name, role_id);
CREATE        INDEX webacula_client_acl_idx2 ON webacula_client_acl (id, order_acl);



CREATE TABLE webacula_command_acl (
    id       SERIAL NOT NULL,
    dt_id    integer,
    role_id   integer,
    PRIMARY KEY (id)
);
CREATE UNIQUE INDEX webacula_command_acl_idx1 ON webacula_command_acl (dt_id, role_id);



CREATE TABLE webacula_dt_commands (
    id      SERIAL NOT NULL,
    name    varchar(127) UNIQUE not null,
    description TEXT NOT NULL,
    primary key (id)
);



-- see src/dird/ua_cmds.c
INSERT INTO webacula_dt_commands (id, name, description) VALUES
    (1,   '*all*',       'All commands'),
    (10,  'add',         'Add media to a pool'),
    (20,  'autodisplay', 'Autodisplay console messages'),
    (30,  'automount',   'Automount after label'),
    (40,  'cancel',      'Cancel a job'),
    (50,  'create',      'Create DB Pool from resource'),
    (60,  'delete',      'Delete volume, pool or job'),
    (70,  'disable',     'Disable a job'),
    (80,  'enable',      'Enable a job'),
    (90,  'estimate',    'Performs FileSet estimate, listing gives full listing'),
    (100, 'exit',        'Terminate Bconsole session'),
    (110, 'gui',         'Non-interactive gui mode'),
    (120, 'help',        'Print help on specific command'),
    (130, 'label',       'Label a tape'),
    (140, 'list',        'List objects from catalog'),
    (150, 'llist',       'Full or long list like list command'),
    (160, 'messages',    'Display pending messages'),
    (170, 'memory',      'Print current memory usage'),
    (180, 'mount',       'Mount storage'),
    (190, 'prune',       'Prune expired records from catalog'),
    (200, 'purge',       'Purge records from catalog'),
    (210, 'python',      'Python control commands'),
    (220, 'quit',        'Terminate Bconsole session'),
    (230, 'query',       'Query catalog'),
    (240, 'restore',     'Restore files'),
    (250, 'relabel',     'Relabel a tape'),
    (260, 'release',     'Release storage'),
    (270, 'reload',      'Reload conf file'),
    (280, 'run',         'Run a job'),
    (290, 'status',      'Report status'),
    (300, 'setdebug',    'Sets debug level'),
    (310, 'setip',       'Sets new client address, if authorized'),
    (320, 'show',        'Show resource records'),
    (330, 'sqlquery',    'Use SQL to query catalog'),
    (340, 'time',        'Print current time'),
    (350, 'trace',       'Turn on/off trace to file'),
    (360, 'unmount',     'Unmount storage'),
    (370, 'umount',      'Umount - for old-time Unix guys, see unmount'),
    (380, 'update',      'Update volume, pool or stats'),
    (390, 'use',         'Use catalog xxx'),
    (400, 'var',         'Does variable expansion'),
    (410, 'version',     'Print Director version'),
    (420, 'wait',        'Wait until no jobs are running');



CREATE TABLE webacula_fileset_acl (
    id        SERIAL NOT NULL,
    name      varchar(127),
    order_acl integer,
    role_id   integer,
    PRIMARY KEY (id)
);
CREATE UNIQUE INDEX webacula_fileset_acl_idx1 ON webacula_fileset_acl (name, role_id);
CREATE        INDEX webacula_fileset_acl_idx2 ON webacula_fileset_acl (id, order_acl);



CREATE TABLE webacula_job_acl (
    id        SERIAL NOT NULL,
    name      varchar(127),
    order_acl integer,
    role_id   integer,
    PRIMARY KEY (id)
);
CREATE UNIQUE INDEX webacula_job_acl_idx1 ON webacula_job_acl (name, role_id);
CREATE        INDEX webacula_job_acl_idx2 ON webacula_job_acl (id, order_acl);



CREATE TABLE webacula_pool_acl (
    id        SERIAL NOT NULL,
    name      varchar(127),
    order_acl integer,
    role_id   integer,
    PRIMARY KEY (id)
);
CREATE UNIQUE INDEX webacula_pool_acl_idx1 ON webacula_pool_acl (name, role_id);
CREATE        INDEX webacula_pool_acl_idx2 ON webacula_pool_acl (id, order_acl);



CREATE TABLE webacula_storage_acl (
    id        SERIAL NOT NULL,
    name      varchar(127),
    order_acl integer,
    role_id   integer,
    PRIMARY KEY (id)
);
CREATE UNIQUE INDEX webacula_storage_acl_idx1 ON webacula_storage_acl (name, role_id);
CREATE        INDEX webacula_storage_acl_idx2 ON webacula_storage_acl (id, order_acl);



CREATE TABLE webacula_where_acl (
    id        SERIAL NOT NULL,
    name      TEXT NOT NULL,
    order_acl integer,
    role_id   integer,
    PRIMARY KEY (id)
);
CREATE UNIQUE INDEX webacula_where_acl_idx1 ON webacula_where_acl (substring(name from 0 for 256), role_id);
CREATE        INDEX webacula_where_acl_idx2 ON webacula_where_acl (id, order_acl);



-- 'root_role' Bacula ACLs
INSERT INTO webacula_storage_acl (name, order_acl, role_id)  VALUES ('*all*', 1, 1);
INSERT INTO webacula_pool_acl    (name, order_acl, role_id)  VALUES ('*all*', 1, 1);
INSERT INTO webacula_client_acl  (name, order_acl, role_id)  VALUES ('*all*', 1, 1);
INSERT INTO webacula_fileset_acl (name, order_acl, role_id)  VALUES ('*all*', 1, 1);
INSERT INTO webacula_where_acl   (name, order_acl, role_id)  VALUES ('*all*', 1, 1);
INSERT INTO webacula_command_acl (dt_id,role_id) VALUES (1, 1);
INSERT INTO webacula_job_acl     (name, order_acl, role_id)  VALUES ('*all*', 1, 1);

-- 'operator_role' Bacula ACLs
INSERT INTO webacula_storage_acl (name, order_acl, role_id)  VALUES ('*all*', 1, 2);
INSERT INTO webacula_pool_acl    (name, order_acl, role_id)  VALUES ('*all*', 1, 2);
INSERT INTO webacula_client_acl  (name, order_acl, role_id)  VALUES ('*all*', 1, 2);
INSERT INTO webacula_fileset_acl (name, order_acl, role_id)  VALUES ('*all*', 1, 2);
INSERT INTO webacula_where_acl   (name, order_acl, role_id)  VALUES ('*all*', 1, 2);
INSERT INTO webacula_command_acl (dt_id,role_id) VALUES (1, 2);
INSERT INTO webacula_job_acl     (name, order_acl, role_id)  VALUES ('*all*', 1, 2);


-- PHP session storage
CREATE TABLE webacula_php_session (
    id       char(128),
    modified integer,
    lifetime integer,
    data_session TEXT,
    login    varchar(50),
    PRIMARY KEY (id)
);


END-OF-DATA

res=$?
if test $res = 0; then
   echo "PostgreSql : create of Webacula ACL tables succeeded."
else
   echo "PostgreSql : create of Webacula ACL tables failed!"
fi
exit $res
